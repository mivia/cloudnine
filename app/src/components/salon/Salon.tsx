import React from 'react';

export const Salon: React.FC = () => {
  return <h2>Salon</h2>;
};
