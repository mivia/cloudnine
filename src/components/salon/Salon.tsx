import React from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { RouteComponentProps } from 'react-router';

import {
  List,
  ListItem,
  Grid,
  Divider,
  Typography,
  IconButton,
  Paper,
  Tabs,
  Tab,
  Box,
} from '@material-ui/core';

import data from '../../data/salons.json';
import smallImage from '../../assets/images/image.png';
import mediumImage from '../../assets/images/image@2x.png';
import largeImage from '../../assets/images/image@3x.png';
import PinIcon from '../../assets/icons/pin.svg';
import ClockIcon from '../../assets/icons/clock.svg';
import PhoneIcon from '../../assets/icons/phone.svg';
import GlobeIcon from '../../assets/icons/globe.svg';
import DownArrowIcon from '../../assets/icons/down-arrow.svg';

import LeftArrowIcon from '../../assets/icons/left-arrow-white.svg';
import HeartIcon from '../../assets/icons/heart.svg';

import SalonNameAndRating from './SalonNameAndRating/SalonNameAndRating';

const useStyles = makeStyles((theme) => ({
  imageContainer: {
    height: '40vh',
    color: 'white',
    display: 'flex',
    position: 'relative',
    background: 'linear-gradient(to bottom, transparent 50%, black 100%)',
  },
  nameAndRatingContainer: {
    position: 'absolute',
    left: 15,
    bottom: 30,
    '& h5': {
      fontSize: 22,
    },
    '& h6': {
      color: 'white',
    },
    '& span': {
      fontSize: 20,
    },
  },
  img: {
    width: '100%',
    objectFit: 'cover',
    zIndex: -1,
  },
  tabs: {
    '& span': {
      fontSize: 15,
      color: 'black',
      textTransform: 'capitalize',
    },
    '& button': {
      maxWidth: 'none',
      flex: '50%',
    },
  },
  dividerBox: {
    height: 25,
    backgroundColor: '#f9f9f9',
  },
  infoList: {
    '& li': {
      padding: '14px 15px',
    },
    '& img': {
      marginRight: theme.spacing(2),
    },
    '& hr': {
      width: 'calc(100% - 30px)',
    },
  },
  arrowIcon: {
    marginLeft: theme.spacing(2),
  },
  backIcon: {
    position: 'absolute',
    left: 15,
  },
  heartIcon: {
    position: 'absolute',
    right: 15,
  },
}));

export interface ISalon {
  id: number;
  name: string;
  rating: number;
  votes: number;
  price: number;
  address: string;
  topLeftNumber: number;
  duration: number;
}

type Props = RouteComponentProps<{
  id: string;
}>;

export const Salon: React.FC<Props> = (props) => {
  const {
    match: {
      params: { id: salonId },
    },
  } = props;

  const classes = useStyles();
  const history = useHistory();

  const salonToDisplay = data.salons.find(
    (salon) => salon.id === Number(salonId),
  );

  return (
    <>
      {salonToDisplay && (
        <Grid container direction="column">
          <Grid item className={classes.imageContainer}>
            <img
              className={classes.img}
              srcSet={`${smallImage} 375w,
              ${mediumImage} 750w,
              ${largeImage} 1125w`}
              sizes="(max-width: 600px) 375px, 
              100vw 
            "
            />
            <SalonNameAndRating
              className={classes.nameAndRatingContainer}
              salon={salonToDisplay}
            />
            <IconButton
              onClick={() => history.goBack()}
              aria-label="back"
              className={classes.backIcon}
            >
              <img src={LeftArrowIcon} alt="back" />
            </IconButton>
            <IconButton aria-label="heart-icon" className={classes.heartIcon}>
              <img src={HeartIcon} alt="like-icon" />
            </IconButton>
          </Grid>
          <Grid item>
            <Paper square>
              <Tabs
                value={0}
                indicatorColor="primary"
                textColor="primary"
                aria-label="salon tabs"
                className={classes.tabs}
              >
                <Tab label="Info" />
                <Tab label="Schema" disabled />
              </Tabs>
            </Paper>
          </Grid>
          <Box className={classes.dividerBox} />
          <Grid item>
            <List className={classes.infoList} aria-label="salon-info-list">
              <ListItem>
                <img src={PinIcon} alt="pin-icon" />
                <Typography>Rådmansgatan 46, 113 57 Stockholm</Typography>
              </ListItem>
              <Box justifyContent="center" display="flex">
                <Divider light />
              </Box>
              <ListItem>
                <img src={ClockIcon} alt="clock-icon" />
                <Typography>Öppet till 19:00 idag</Typography>
                <img
                  src={DownArrowIcon}
                  alt="down-arrow-icon"
                  className={classes.arrowIcon}
                />
              </ListItem>
              <Box justifyContent="center" display="flex">
                <Divider light />
              </Box>
              <ListItem>
                <img src={PhoneIcon} alt="phone-icon" />
                <Typography>08-522 389 20</Typography>
              </ListItem>
              <Box justifyContent="center" display="flex">
                <Divider light />
              </Box>
              <ListItem>
                <img src={GlobeIcon} alt="globe-icon" />
                <Typography>www.salongweb.se</Typography>
              </ListItem>
              <Box justifyContent="center" display="flex">
                <Divider light />
              </Box>
              <ListItem>
                <Typography>
                  Lorem ipsum dolor sit amet, vulputate nunc. Auctor viverra.
                  Ridiculus feugiat nunc porttitor volut pat, acu quis torquent
                  iaculis ultricies massa, duis nun quis que amet.
                </Typography>
              </ListItem>
            </List>
          </Grid>
        </Grid>
      )}
    </>
  );
};
