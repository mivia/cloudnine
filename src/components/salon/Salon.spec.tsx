import React from 'react';
import { Salon } from './Salon';
import renderer from 'react-test-renderer';
import { match } from 'react-router';
import { createMemoryHistory, createLocation } from 'history';

const path = `/route/:id`;

const getRouteComponentPropsMock = (salonId: string) => {
  const match: match<{ id: string }> = {
    isExact: false,
    path,
    url: path.replace(':id', '1'),
    params: { id: salonId },
  };

  return {
    history: createMemoryHistory(),
    location: createLocation(match.url),
    match,
  };
};

describe('<Salon/>', () => {
  it('renders Salon a salon when found', () => {
    const tree = renderer
      .create(<Salon {...getRouteComponentPropsMock('1')}></Salon>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders nothing when a salon is not found', () => {
    const tree = renderer
      .create(<Salon {...getRouteComponentPropsMock('11')}></Salon>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
