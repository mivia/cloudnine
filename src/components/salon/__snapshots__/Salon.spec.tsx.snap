// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Salon/> renders Salon a salon when found 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root makeStyles-imageContainer-1 MuiGrid-item"
  >
    <img
      className="makeStyles-img-3"
      sizes="(max-width: 600px) 375px,  100vw  "
      srcSet="image.png 375w,
              image@2x.png 750w,
              image@3x.png 1125w"
    />
    <div
      className="makeStyles-nameAndRatingContainer-2"
    >
      <h5
        className="MuiTypography-root makeStyles-heading-11 MuiTypography-h5"
      >
        salon 1
      </h5>
      <div
        className="MuiBox-root MuiBox-root-12"
      >
        <span
          aria-label="4 Stars"
          className="MuiRating-root makeStyles-rating-10 MuiRating-readOnly"
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="img"
        >
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
          (50)
        </h6>
      </div>
    </div>
    <button
      aria-label="back"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-backIcon-8"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt="back"
          src="left-arrow-white.svg"
        />
      </span>
    </button>
    <button
      aria-label="heart-icon"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-heartIcon-9"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt="like-icon"
          src="heart.svg"
        />
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1"
    >
      <div
        className="MuiTabs-root makeStyles-tabs-4"
      >
        <div
          className="MuiTabs-scroller MuiTabs-fixed"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": null,
              "overflow": "hidden",
            }
          }
        >
          <div
            aria-label="salon tabs"
            className="MuiTabs-flexContainer"
            onKeyDown={[Function]}
            role="tablist"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Info
              </span>
              <span
                className="PrivateTabIndicator-root-13 PrivateTabIndicator-colorPrimary-14 MuiTabs-indicator"
                style={Object {}}
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                Schema
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-17 makeStyles-dividerBox-5"
  />
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <ul
      aria-label="salon-info-list"
      className="MuiList-root makeStyles-infoList-6 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <img
          alt="pin-icon"
          src="pin.svg"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Rådmansgatan 46, 113 57 Stockholm
        </p>
      </li>
      <div
        className="MuiBox-root MuiBox-root-18"
      >
        <hr
          className="MuiDivider-root MuiDivider-light"
        />
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <img
          alt="clock-icon"
          src="clock.svg"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Öppet till 19:00 idag
        </p>
        <img
          alt="down-arrow-icon"
          className="makeStyles-arrowIcon-7"
          src="down-arrow.svg"
        />
      </li>
      <div
        className="MuiBox-root MuiBox-root-19"
      >
        <hr
          className="MuiDivider-root MuiDivider-light"
        />
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <img
          alt="phone-icon"
          src="phone.svg"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          08-522 389 20
        </p>
      </li>
      <div
        className="MuiBox-root MuiBox-root-20"
      >
        <hr
          className="MuiDivider-root MuiDivider-light"
        />
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <img
          alt="globe-icon"
          src="globe.svg"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          www.salongweb.se
        </p>
      </li>
      <div
        className="MuiBox-root MuiBox-root-21"
      >
        <hr
          className="MuiDivider-root MuiDivider-light"
        />
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Lorem ipsum dolor sit amet, vulputate nunc. Auctor viverra. Ridiculus feugiat nunc porttitor volut pat, acu quis torquent iaculis ultricies massa, duis nun quis que amet.
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Salon/> renders nothing when a salon is not found 1`] = `null`;
